polygeist/lib/Dialect/Polygeist/Transforms/LICM.cpp

```diff
@@ -703,16 +703,44 @@ void AffineParallelGuardBuilder::getConstraints(
     }
 }
 
+//===----------------------------------------------------------------------===//
+// LICMCandidate
+//===----------------------------------------------------------------------===//
+
+class LICMCandidate {
+public:
+  LICMCandidate(Operation &op) : op(op) {}
+  Operation &getOp() const { return op; }
+  void addDependee(Operation &op) { dependees.push_back(&op); }
+
+private:
+  Operation &op;
+  SmallVector<Operation *> dependees;
```

Comment:
`dependees` -> prerequisites; // Operations that need to be hoisted to allow this operation to be hoisted.